lib/publiccloud/gcp_client.pm

```diff
@@ -71,7 +72,27 @@ sub get_next_vault_role {
 sub create_credentials_file {
     my ($self) = @_;
     my $credentials_file;
+    if (get_var('PUBLIC_CLOUD_CREDENTIALS_URL')) {
+        my $json = get_credentials();
+        # Need to remove quotes, convert \n to \\n and / to \/
+        assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\\"//g' | sed 's/\\\n/\\\\\\\n/g' | sed 's/\\//\\\\\//g')});
+        #assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\"//g' | sed 's/\\n/\\\\n/g'      | sed 's/\//\\\//g')});
+
+        $self->private_key_id(get_required_var('PUBLIC_CLOUD_PRIVATE_KEY_ID'));
+        $credentials_file = "{" . $/
```

Comment:
Same as above, wouldn't it make much more sense to add those attributes to the `gce.json` served as credentials file or do we need to assemble it here, because some vars (project_id, ...) need to differ?

Reply:
I would prefer that the service returns all the fields, and we avoid adding all those variables